src/thread/thread_entry.hpp

```diff
@@ -304,9 +312,14 @@ namespace cubthread
 
       void return_lock_free_transaction_entries (void);
 
+    private:
+      friend void ::thread_lock_entry (::cubthread::entry *thread_p);
```

Comment:
These changes helped during investigation for this PR in the following way: I wanted to make sure that the `lock`, `unlock` function are never called directly. Thus, I made them private and allowed `::thread_lock_entry` and `::thread_unlock_entry` as friend functions to be able to call them.
Are unrelated to the rest of the PR and I am willing to split them to a separate PR if so advised by reviewers.